### DISCUSSION TITLE
Reconnect and messages sent to the client while disconnected

### DISCUSSION THREAD

**Discussion by Clonex** [socketio/socket.io#4570] [Q&A] [answered]
Hello,
We are considering switching our game server to use Socket.IO.
But I cant seem to find any documentation about how the reconnect works in practice.

My question is, does the library queue up messages sent from the server to a disconnected client. 
So it can re-emmit them when the client is reconnected? Or will we have to implement that ourselves?

**Comment by darrachequesne** [comment, marked as answer]
> My question is, does the library queue up messages sent from the server to a disconnected client.

No, there is currently no buffer on the server.

Reference: https://socket.io/docs/v4/delivery-guarantees

There are ongoing discussions about adding that feature though: https://github.com/socketio/socket.io/discussions/4510

**Reply by Clonex** [reply]
I see, thanks for clearing it up!
